transform.ml

```diff
@@ -130,22 +139,17 @@ let optimize (opts : string list) (prog : program) : program option =
     | o ->
       raise (UnknownOptimization o)
   in
-  let opts_assumpt, opts_other = List.partition (fun o -> List.mem o assumption_opts) opts in
   let optimizers = (List.map optimizer opts) @ [(as_opt_program cleanup_all)] in
-  let optimizers_classic = (List.map optimizer opts_other) @ [(as_opt_program cleanup_all)] in
   let optimizer = combine_opt optimizers in
-  let optimizer_classic = combine_opt optimizers_classic in
 
   let optimizer =
-    if opts_assumpt <> []
-    then combine_opt [
+    combine_opt [
         (as_opt_program Transform_assumption.insert_checkpoints);
         optimizer;
-        (as_opt_program (as_opt_function Transform_assumption.hoist_assumption));
         optimizer;
         Transform_assumption.remove_empty_osr;
-        optimizer_classic;
+        optimizer;
```

Comment:
no good answer...

some transformations might enable other, so running multiple times makes more optimizations.

maybe we could instead have a cli argument for it...